corehq/apps/reports/standard/monitoring.py

```diff
@@ -181,15 +184,33 @@ class CaseActivityReport(WorkerMonitoringCaseReportTableBase):
     """See column headers for details"""
     name = gettext_lazy('Case Activity')
     slug = 'case_activity'
-    fields = ['corehq.apps.reports.filters.users.ExpandedMobileWorkerFilter',
-              'corehq.apps.reports.filters.select.CaseTypeFilter']
     display_data = ['percent']
     emailable = True
     description = gettext_lazy("Followup rates on active cases.")
     is_cacheable = True
     ajax_pagination = True
     exportable_all = True
 
+    @property
+    def fields(self):
+        fields = [
+            'corehq.apps.reports.filters.users.ExpandedMobileWorkerFilter',
+            'corehq.apps.reports.filters.select.CaseTypeFilter',
+        ]
+        if self.has_case_sharing:
+            fields.append('corehq.apps.reports.filters.users.UserOrGroupFilter')
+        return fields
+
+    @property
+    @memoized
+    def has_case_sharing(self):
+        return self.domain_object.case_sharing_included()
+
+    @property
+    @memoized
+    def view_by_groups(self):
+        return self.request.GET.get('view_by', None) == 'groups'
```

Comment:
nit: isn't the default value already `None`?

Reply:
This is true. I put this in for better readability on what the default is (also done in other places such as [here](https://github.com/dimagi/commcare-hq/blob/0af7ed0eb24093fe68c04141dccf616c0e51c1ef/corehq/apps/app_manager/views/cli.py#L73)). No strong feelings with keeping this format however, so I'll remove the default value.

Reply:
[c0293ae](https://github.com/dimagi/commcare-hq/pull/34464/commits/c0293ae0e09175607011de1a3114ad03186553d5)